### PR TITLE
Support Node 8

### LINKED ISSUE #2637: Bring back node 8 support by transpiling async functions
`ncc` by default only transpiles async function into generator functions. Node 8 can be supported by transpiling into promise chains https://www.npmjs.com/package/babel-plugin-transform-async-to-promises.

Also add it to the CircleCI version matrix.

Related to https://github.com/zeit/now-cli/pull/2603.

### PR DESCRIPTION
**Pull request by TooTallNate**
This changes the TypeScript compilation target to "es2015" so that async generator functions are transpiled, which enables Node 8 to be supported.

As a side-effect, any of the files that utilize `async function*` or `for await` needed to be converted to TypeScript so that the transpilation actually happens.

It was painful to do this process for `src/commands/deploy/legacy.ts`, and since it is legacy code I did not want to make too many changes to the code, so there's a log of `@ts-ignore` and `any` types being used. But the behavior of the file should not have changed at all.

Closes #2637.

### CODE REVIEW

File: src/util/deploy/get-state-change-from-polling.ts [hunk 1]
```diff
@@ -8,10 +8,10 @@ import getDeploymentByIdOrThrow from './get-deployment-by-id-or-throw';
 const POLLING_INTERVAL = 5000;
 
 async function* getStatusChangeFromPolling(
-  now,
-  contextName,
-  idOrHost,
-  initialState
+  now: any,
```
Comment (AndyBitz) [inline src/util/deploy/get-state-change-from-polling.ts:11]: I think it's possible to just `import Now from './src/util'` and use it as type here

File: package.json [hunk 2]
```diff
@@ -190,6 +190,7 @@
     "universal-analytics": "0.4.20",
     "update-check": "1.5.3",
     "utility-types": "2.1.0",
+    "whatwg-url": "7.0.0",
```
Comment (AndyBitz) [inline package.json]: Why exactly is this needed, instead of just using the native URL module?

Comment (TooTallNate) [reply package.json]: Native `URL` is not available in Node 8

File: src/util/deploy/get-state-change-from-polling.ts (hunk 1 above)
Comment (TooTallNate) [reply src/util/deploy/get-state-change-from-polling.ts:11]: Good call. See 57290bab.

File: package.json (hunk 2 above)
Comment (AndyBitz) [reply package.json]: @TooTallNate really?

image

Comment (TooTallNate) [reply package.json]: Oh good catch. It's the _global_ `URL` object that is not available in Node 8, but we can import this one and use it 👍

Comment (TooTallNate) [reply package.json]: ebfbbd42

**Review by javivelasco — APPROVED** [review]

**Review by AndyBitz — APPROVED** [review]

**Pull request merged**